com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/TimeoutAsyncFlowableStep.java

```diff
@@ -28,9 +28,19 @@ private boolean hasTimedOut(DelegateExecution context) {
     private long getStepStartTime(DelegateExecution context) {
         Long stepStartTime = (Long) context.getVariable(getStepStartTimeVariable());
         if (stepStartTime == null) {
-            stepStartTime = System.currentTimeMillis();
-            context.setVariable(getStepStartTimeVariable(), stepStartTime);
+            return retrieveAndSetStartTime(context);
         }
+
+        if (StepsUtil.getStepPhase(context) == StepPhase.RETRY) {
+            return retrieveAndSetStartTime(context);
+        }
+
+        return stepStartTime;
+    }
+
+    private Long retrieveAndSetStartTime(DelegateExecution context) {
```

Comment:
Generally, methods with "and" in their name should be refactored, because they do more than one thing (SRP).

Reply:
Correct. Will rename it